src/gui/mzroll/database.cpp

```diff
@@ -907,9 +896,36 @@ int Database::loadCompoundCSVFile(string filename)
                 id = name;
             invalidRows.push_back(id);
         }
+        sort(compoundsDB.begin(),compoundsDB.end(), Compound::compMass);
```

Comment:
This call to `sort` should happen once at the end, once all compounds have been read. Right now it will be called everytime this lambda is invoked. So, it gets worse with increasing number of entries in CSV database.